ipam/allocator.go

```diff
@@ -264,6 +251,33 @@ func (alloc *Allocator) free(ident string) error {
 	return <-errChan
 }
 
+// Free (Sync) - release single IP address for container
+func (alloc *Allocator) Free(ident string, addr address.Address) error {
+	errChan := make(chan error)
+	alloc.actionChan <- func() {
+		addrs := alloc.owned[ident]
+		for i := range addrs {
+			if addrs[i] == addr {
```

Comment:
I doubt the compiler figures that out. To me it looks confusing to use a single-return `range` when the very first expression in the block extracts the value.